### PR TITLE
Fix warnings on the latest nightly

### PR DESCRIPTION
**Pull request by taiki-e**
This fixes warnings due to new clippy lints added by https://github.com/rust-lang/rust-clippy/pull/3766.

### CODE REVIEW

**Comment by taiki-e** [comment]
Oh, there also seem to be warnings due to `incomplete_features` became lint by https://github.com/rust-lang/rust/pull/63095.

**Comment by taiki-e** [comment]
😕
```text
$ rustup target add thumbv7m-none-eabi
info: downloading component 'rust-std' for 'thumbv7m-none-eabi'
error: component download failed for rust-std-thumbv7m-none-eabi
info: caused by: could not download file from 'https://static.rust-lang.org/dist/2019-08-02/rust-std-nightly-thumbv7m-none-eabi.tar.xz' to '/home/travis/.rustup/downloads/ac33d8fd5a4b68735a9762b26807ab273e8f155557002c5388dddd5e22335cb7.partial'
info: caused by: failed to make network request
info: caused by: https://static.rust-lang.org/dist/2019-08-02/rust-std-nightly-thumbv7m-none-eabi.tar.xz: timed out
The command "rustup target add thumbv7m-none-eabi" failed and exited with 1 during .
```

File: futures-util/src/compat/compat01as03.rs [hunk 1]
```diff
@@ -413,6 +414,7 @@ mod io {
         ///
         /// ```
         /// #![feature(async_await, impl_trait_in_bindings)]
+        /// # #![allow(incomplete_features)]
```
Comment (cramertj) [inline futures-util/src/compat/compat01as03.rs]: I don't see where this example uses the `impl_trait_in_bindings` feature

Comment (taiki-e) [reply futures-util/src/compat/compat01as03.rs]: Oops, I missed that. Removed

**Comment by cramertj** [comment]
Great, thanks!

**Pull request merged**